src/wp-includes/rest-api/endpoints/class-wp-rest-abilities-run-controller.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ * REST API run controller for Abilities API.
+ *
+ * @package WordPress
+ * @subpackage Abilities_API
+ * @since 6.9.0
+ */
+
+declare( strict_types = 1 );
+
+/**
+ * Core controller used to execute abilities via the REST API.
+ *
+ * @since 6.9.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Abilities_Run_Controller extends WP_REST_Controller {
+
+	/**
+	 * REST API namespace.
+	 *
+	 * @since 6.9.0
+	 * @var string
+	 */
+	protected $namespace = 'wp/v2';
+
+	/**
+	 * REST API base route.
+	 *
+	 * @since 6.9.0
+	 * @var string
+	 */
+	protected $rest_base = 'abilities';
+
+	/**
+	 * Registers the routes for ability execution.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>[a-zA-Z0-9\-\/]+?)/run',
```

Comment:
Why wouldn't underscores be allowed?

```suggestion
			'/' . $this->rest_base . '/(?P<name>[a-zA-Z0-9\-\/_]+?)/run',
```

Reply:
`_` isn't allowed in the ability name, so it propagates to the route name.